Improvements to app documentation (#31)
Esri/nearby-javascript: RELEASE.md

> # Release 1.0.2

Reviewer: We may not have these headers consistent from one OSA repo to another but I do think they should remain consistent within each project.
```suggestion
## 1.0.2
```

Author: Very good catch!

- Adds doc table of contents to root README.md and docs/index.md
- Renames docs/index.md to [docs/README.md](/docs/README.md)

## 1.0.1

> - Comprehensive [app documentation](/documentation/README.md) from the ArcGIS for Developers site.

Reviewer: ```suggestion
- Comprehensive [app documentation](/docs/README.md) from the ArcGIS for Developers site.
```

Author: Oof, don't know how I missed that. Thanks, just corrected.

> - Comprehensive [app documentation](/documentation/index.md) from the ArcGIS for Developers site.

Reviewer: ```suggestion
- Comprehensive [app documentation](/docs/README.md) from the ArcGIS for Developers site.
```
